MantidPlot/src/Mantid/InstrumentWidget/InstrumentWindow.cpp

```diff
@@ -848,32 +69,31 @@ void InstrumentWindow::preDeleteHandle(
 void InstrumentWindow::afterReplaceHandle(
     const std::string &wsName, const boost::shared_ptr<Workspace> workspace) {
   // Replace current workspace
-  if (wsName == m_workspaceName.toStdString()) {
-    if (m_instrumentActor) {
+  if (wsName == m_instrumentWidget->getWorkspaceName().toStdString()) {
+    if (m_instrumentWidget->getInstrumentActor()) {
       // Check if it's still the same workspace underneath (as well as having
       // the same name)
       auto matrixWS =
           boost::dynamic_pointer_cast<const MatrixWorkspace>(workspace);
       bool sameWS = false;
       try {
-        sameWS = (matrixWS == m_instrumentActor->getWorkspace());
+        sameWS = (matrixWS ==
+                  m_instrumentWidget->getInstrumentActor()->getWorkspace());
       } catch (std::runtime_error &) {
         // Carry on, sameWS should stay false
       }
 
       // try to detect if the instrument changes (unlikely if the workspace
       // hasn't, but theoretically possible)
-      bool resetGeometry = matrixWS->getInstrument()->getNumberDetectors() !=
-                           m_instrumentActor->ndetectors();
+      bool resetGeometry =
+          matrixWS->getInstrument()->getNumberDetectors() !=
+          m_instrumentWidget->getInstrumentActor()->ndetectors();
 
       // if workspace and instrument don't change keep the scaling
       if (sameWS && !resetGeometry) {
-        m_instrumentActor->updateColors();
+        m_instrumentWidget->getInstrumentActor()->updateColors();
```

Comment:
Same thing as above double accessor implies a method is missing on `InstrumentWidget`